src/dune_engine/build_system.ml

```diff
@@ -1458,10 +1458,10 @@ end = struct
         ; attached_to_alias : bool
         }
 
-  let targets_without_stamp_file targets = function
-    | Normal_rule -> targets
+  let targets_without_stamp_file files = function
```

Comment:
An alternative name for this function could be `remove_stamp_file` (since the argument can be any set of files).

Reply:
That seems better indeed, changing it